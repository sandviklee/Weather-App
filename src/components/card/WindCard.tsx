import styles from "./WindCard.module.css";

interface CardProps {
  WindSpeed: number;
  WindGust: number;
  WindDirection: number;
}

const convertFromDegreesToDirectionString = (degrees: number) => {
  const directions = [
    "nord",
    "nordøst",
    "øst",
    "sørøst",
    "sør",
    "sørvest",
    "vest",
    "nordvest",
  ];
  const index = Math.round(degrees / 45) % 8;
  return directions[index];
};

const UVCard = ({ WindSpeed, WindGust, WindDirection }: CardProps) => {
  return (
    <div className={styles.container}>
      <p className={styles.title}>Vind</p>
      <div className={styles.info_container}>
        <p className={styles.uv_index}>
          {WindSpeed}m/s{" "}
          <span style={{ color: "rgba(60,103,255,0.6)" }}> ({WindGust})</span>
        </p>
        <div className={styles.compass}>
          {convertFromDegreesToDirectionString(WindDirection)}
          <img
            src="/project1/icons/compass.svg"
            alt="arrow"
            width="24px"
            height="24px"
            style={{ transform: `rotate(${WindDirection}deg)` }}
          />
        </div>
      </div>
    </div>
  );
};

export default UVCard;
