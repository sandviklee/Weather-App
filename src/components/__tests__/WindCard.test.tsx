import { render, screen } from '@testing-library/react';
import WindCard from '../card/WindCard';

describe('WindCard component', () => {
    test('renders correctly', () => {
        const testWindSpeed = 5;
        const testWindGust = 10;
        const testWindDirection = 90;
        render(<WindCard WindSpeed={testWindSpeed} WindGust={testWindGust} WindDirection={testWindDirection} />);
        
        const windTitle = screen.getByText('Vind');
        expect(windTitle).toBeInTheDocument();
        const windSpeedValue = screen.getByText(`${testWindSpeed}m/s`);
        expect(windSpeedValue).toBeInTheDocument();
        const windGustValue = screen.getByText(`(${testWindGust})`);
        expect(windGustValue).toBeInTheDocument();
        const windDirectionValue = screen.getByText('Ã¸st');
        expect(windDirectionValue).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { asFragment } = render(<WindCard WindSpeed={5} WindGust={10} WindDirection={90} />);
        expect(asFragment()).toMatchSnapshot();
    });
});