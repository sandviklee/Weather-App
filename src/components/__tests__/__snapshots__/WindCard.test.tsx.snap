// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WindCard component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_1d4056"
  >
    <p
      class="_title_1d4056"
    >
      Vind
    </p>
    <div
      class="_info_container_1d4056"
    >
      <p
        class="_uv_index_1d4056"
      >
        5m/s 
        <span
          style="color: rgba(60, 103, 255, 0.6);"
        >
           (10)
        </span>
      </p>
      <div
        class="_compass_1d4056"
      >
        Ã¸st
        <img
          alt="arrow"
          height="24px"
          src="/project1/icons/compass.svg"
          style="transform: rotate(90deg);"
          width="24px"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
